pkg/infra/iac2/templates/helm_chart_alb_controller/factory.ts

```diff
@@ -0,0 +1,61 @@
+// A specialization of the more generic helm_chart/ template, for the ALB controller. This needs to create a service
+// account, which itself needs things like VPC id and a role, so it's easier to just create it as a separate template.
+import * as aws from '@pulumi/aws'
+import * as pulumi from '@pulumi/pulumi'
+import * as pulumi_k8s from '@pulumi/kubernetes'
+
+interface Args {
+    Name: string
+    Namespace: string
+    ClustersProvider: pulumi_k8s.Provider
+    ClusterName: string
+    Region: pulumi.Output<pulumi.UnwrappedObject<aws.GetRegionResult>>
+    Vpc: aws.ec2.Vpc
+    Role: aws.iam.Role
+    dependsOn: pulumi.Input<pulumi.Input<pulumi.Resource>[]> | pulumi.Input<pulumi.Resource>
+}
+
+// noinspection JSUnusedLocalSymbols
+function create(args: Args): pulumi_k8s.helm.v3.Chart {
```

Comment:
why a separate factory?

Reply:
I thought it makes for a simpler way to get the various nested values into the `values` map, especially with all the `pulumi.interpolate` that needs to happen. It seemed easier than having to create new IacValues, though I could do that if you prefer.

Reply:
well we have the Arn, name and id iac value i think so they should all fit.

We can try to do it after though, would rather get it in so we can start testing, but maybe add a card to fit and finish to clean this up?

Reply:
Alright, I can convert it to use IaCValue.

I assume I'll need to manually set up the DAG dependencies then, right?